crates/ty_python_semantic/resources/mdtest/function/return_type.md

```diff
@@ -466,11 +466,18 @@ def f(cond: bool) -> int:
     return "hello" if cond else NotImplemented
 ```
 
+`NotImplemented` is only special-cased for return types (mirroring the way the interpreter applies
+special casing for the symbol at runtime). It is not generally considered assignable to every other
+type:
+
+```py
+x: int = NotImplemented  # error: [invalid-assignment]
+```
```

Comment:
hmm, mypy/pyright/pyrefly all do not emit a diagnostic here. And the fact that we reject this is the cause of many of the new scipy diagnostics. But it seems clearly unsound to me; I'd see the fact that we reject this as a feature rather than an incompatibility.

Reply:
And I think it's probably just because of the fiction typeshed tells about the MRO of `NotImplementedType`, rather than a deliberately implemented feature of any of those type checkers